packages/@n8n/config/src/configs/logging.config.ts

```diff
@@ -0,0 +1,46 @@
+import { Config, Env, Nested } from '../decorators';
+
+@Config
+class FileLoggingConfig {
+	/**
+	 * Max number of log files to keep, or max number of days to keep logs for.
+	 * Once the limit is reached, the oldest log files will be rotated out.
+	 * If using days, append a `d` suffix. Only for `file` log output.
+	 *
+	 * @example `N8N_LOG_FILE_COUNT_MAX=7` will keep at most 7 files.
+	 * @example `N8N_LOG_FILE_COUNT_MAX=7d` will keep at most 7 days worth of files.
+	 */
+	@Env('N8N_LOG_FILE_COUNT_MAX')
+	fileCountMax: number = 100;
+
+	/** Max size (in MiB) for each log file. Only for `file` log output. */
+	@Env('N8N_LOG_FILE_SIZE_MAX')
+	fileSizeMax: number = 16;
+
+	/** Location of the log files inside `~/.n8n`. Only for `file` log output. */
+	@Env('N8N_LOG_FILE_LOCATION')
+	location: string = 'logs/n8n.log';
+}
+
+@Config
+export class LoggingConfig {
+	/**
+	 * Minimum level of logs to output. Logs with this or higher level will be output;
+	 * logs with lower levels will not. Exception: `silent` disables all logging.
+	 *
+	 * @example `N8N_LOG_LEVEL=info` will output `error`, `warn` and `info` logs, but not `debug`.
+	 */
+	@Env('N8N_LOG_LEVEL')
+	level: 'error' | 'warn' | 'info' | 'debug' | 'silent' = 'info';
+
+	/**
+	 * Where to output logs to. Options are: `console` or `file` or both in a comma separated list.
+	 *
+	 * @example `N8N_LOG_OUTPUT=console,file` will output to both console and file.
+	 */
+	@Env('N8N_LOG_OUTPUT')
+	output: string = 'console';
```

Comment:
Just how we created `JsonStringArray`, we could create a `StringArray` class
```typescript
class StringArray<T extends string> extends Array<T> {
	constructor(str: string) {
		super();
		const parsed = str.split(',') as StringArray<T>;
		return isStringArray(parsed) ? parsed : [];
	}
}
```

and use that here like

```suggestion
	output: StringArray<'console' | 'file'>  = ['console'];
```

Reply:
[d3c794a](https://github.com/n8n-io/n8n/pull/11031/commits/d3c794a6192ea03937b61a281d5460dc3f6aee8d)